app/models/test_passage.rb

```diff
@@ -0,0 +1,54 @@
+# frozen_string_literal: true
+
+class TestPassage < ApplicationRecord
+  belongs_to :user
+  belongs_to :test
+  belongs_to :current_question, class_name: 'Question', optional: true
+
+  before_validation :before_validation_set_first_question, on: :create
+
+  def accept!(answer_ids)
+    self.correct_questions += 1 if correct_answer?(answer_ids)
+    self.current_question = next_question
+    save!
+  end
+
+  def completed?
+    current_question.nil?
+  end
+
+  def passed
```

Comment:
метод возвращает булево
по конвенции такие методы пишут с `?` на конце
```suggestion
  def passed?
```